climada/engine/test/test_impact_calc.py

```diff
@@ -101,6 +101,44 @@ def test_apply_cover_to_mat(self):
             imp.todense(), np.array([[0, 0, 1], [0, 1, 0]])
             )
 
+    def test_error_handling_mismatch_haz_type(self):
+        """Test error handling in case hazard type of hazard
+        does not appear in impf_set or exposures"""
+        exp_haz_correct = deepcopy(ENT.exposures)
+        exp_haz_wrong = deepcopy(ENT.exposures)
+        exp_haz_wrong.gdf.rename(columns={'impf_TC': 'impf_WS'}, inplace=True)
+        impf = ImpactFunc()
+        impf.id = 1
+        impf.intensity = np.array([0, 20])
+        impf.paa = np.array([0, 1])
+        impf.mdd = np.array([0, 0.5])
+        impf.haz_type = 'TC'
+        impfset_haz_correct = ImpactFuncSet()
+        impfset_haz_correct.append(impf)
+        try:
+            ImpactCalc(exp_haz_wrong, impfset_haz_correct, HAZ).impact()
+        except Exception as e:
+            self.assertEqual(str(e), "Impact calculation not possible. No impact "
+                             "functions found for hazard type TC in exposures.")
```

Comment:
Make also an empty hazard.
```suggestion
        haz_tc = Hazard('TC')
        try:
            ImpactCalc(exp_haz_wrong, impfset_haz_correct, HAZ).impact()
        except Exception as e:
            self.assertEqual(str(e), "Impact calculation not possible. No impact "
                             "functions found for hazard type TC in exposures.")
```

Reply:
done